### PR TITLE
MSP-105: Bastion and worker node hardening changes

### PR DESCRIPTION
**Pull request by gavincornwell**
- Now uses a customised version of the bastion bootstrap script to further harden the instances
- Streams all log files from the bastion and worker nodes back to CloudWatch
- Reformatted Fn::Join to Fn::Sub where possible to improve readability
- Added notes on how to run hardening checks

If you have any questions on any of these changes please contact Toni (@toniblyx). Also cc @svidrascu so you have visibility of changes.

### CODE REVIEW

**Comment by gavincornwell** [comment]
Green build on feature branch: https://bamboo.alfresco.com/bamboo/browse/PLAT-ADA44-2

File: templates/bastion-and-eks-cluster.yaml [hunk 1]
```diff
@@ -753,9 +751,10 @@ Resources:
       AutoScalingRollingUpdate:
         MinInstancesInService: !Ref MinNumberOfNodes
         MaxBatchSize: !Ref MaxNumberOfNodes
-    # CreationPolicy:
-    #   ResourceSignal:
-    #     Timeout: PT40M
+    CreationPolicy:
+      ResourceSignal:
+        Count: !Ref DesiredNumberOfNodes
+        Timeout: PT15M
```
Comment (amohammedalfresco) [inline templates/bastion-and-eks-cluster.yaml]: `Timeout: PT15M` seems OK, but may be `25M` can give some extra time.

Comment (gavincornwell) [reply templates/bastion-and-eks-cluster.yaml]: I made it 30M so it's consistent with the bastion configuration.

Comment (amohammedalfresco) [reply templates/bastion-and-eks-cluster.yaml]: Great.  Thanks.

**Pull request merged**